docs/release-process.md

```diff
@@ -32,21 +32,20 @@ To create a new release, follow these steps:
 4. If a problem is found, prepare a fix PR, merge it into the main branch and return to the previous step.
 5. Create a release branch with a name that follows the `release-X.Y` format.
 6. Prepare and merge a PR into the release branch to update the repo files for the release:
-    1. Update the tag of NKG container images used in the installation manifests (both the
-       [deployment manifest](../deploy/manifests/deployment.yaml) and the
-       [provisioner manifest](../conformance/provisioner/provisioner.yaml)),
-       the Helm [values.yaml](../deploy/helm-chart/values.yaml) file, and docs to `X.Y.Z`.
-    2. Ensure that the `imagePullPolicy` is `IfNotPresent` in the installation manifests and the Helm
-       [values.yaml](../deploy/helm-chart/values.yaml) file.
-    3. Modify any `git clone` instructions to use `vX.Y.Z` tag.
-    4. Update the Helm [Chart.yaml](../deploy/helm-chart/Chart.yaml): the `appVersion` to `X.Y.Z`, the icon and source
+    1. Update the Helm [Chart.yaml](../deploy/helm-chart/Chart.yaml): the `appVersion` to `X.Y.Z`, the icon and source
        URLs to point at `vX.Y.Z`, and bump the `version`.
-    5. Update the Helm [README](../deploy/helm-chart/README.md) `--version` flags in the helm commands to use the stable
+    2. Update the Helm [README](../deploy/helm-chart/README.md) `--version` flags in the helm commands to use the stable
        `appVersion` from the previous step.
-    6. Adjust the `VERSION` variable in the [Makefile](../Makefile) and the `NKG_TAG` in the
+    3. Adjust the `VERSION` variable in the [Makefile](../Makefile) and the `NKG_TAG` in the
        [conformance tests Makefile](../conformance/Makefile) to `X.Y.Z`.
-    7. Update the [README](../README.md) to include information about the release.
-    8. Update the [changelog](../CHANGELOG.md). The changelog includes only important (from the user perspective)
+    4. Update the tag of NKG container images used in the Helm [values.yaml](../deploy/helm-chart/values.yaml) file, the
+       [provisioner manifest](../conformance/provisioner/provisioner.yaml)), and all docs to `X.Y.Z`.
+    5. Ensure that the `imagePullPolicy` is `IfNotPresent` in the Helm [values.yaml](../deploy/helm-chart/values.yaml)
+       file.
+    6. Generate the deployment manifests by running `make generate-manifests`.
```

Comment:
since deployment has also another meaning on Kubernetes 
```suggestion
    6. Generate the installation manifests by running `make generate-manifests`.
```